pkg/daemon/update.go

```diff
@@ -81,11 +82,22 @@ func (dn *Daemon) updateOSAndReboot(newConfig *mcfgv1.MachineConfig) error {
 
 		dn.recorder.Eventf(node, corev1.EventTypeNormal, "Drain", "Draining node to update config.")
 
-		err = drain.Drain(dn.kubeClient, []*corev1.Node{node}, &drain.DrainOptions{
-			DeleteLocalData:    true,
-			Force:              true,
-			GracePeriodSeconds: 600,
-			IgnoreDaemonsets:   true,
+		err = wait.ExponentialBackoff(wait.Backoff{
```

Comment:
if this errs, it will still go degraded right?

we need a follow-on item that ensures failure to talk to k8s api never results in degrade of machine.

after all, you can only report degraded by talking to the apiserver.

Reply:
> we need a follow-on item that ensures failure to talk to k8s api never results in degrade of machine.

Hmm, are you suggesting retrying forever?

Reply:
We've been slowly working towards addressing this. Like Colin hinted, it might make sense to retry at a higher level too.